crates/nu-protocol/src/config/mod.rs

```diff
@@ -697,116 +578,91 @@ impl Value {
                             };
                             true
                         })
-                        } else {
-                            report_invalid_value("boolean value is deprecated, should be a record. see `config nu --default`.", span, &mut errors);
-                            // Reconstruct
-                            *value = Value::record(
-                                record! {
-                                    "osc2" => Value::bool(config.shell_integration_osc2, span),
-                                    "ocs7" => Value::bool(config.shell_integration_osc7, span),
-                                    "osc8" => Value::bool(config.shell_integration_osc8, span),
-                                    "osc9_9" => Value::bool(config.shell_integration_osc9_9, span),
-                                    "osc133" => Value::bool(config.shell_integration_osc133, span),
-                                    "osc633" => Value::bool(config.shell_integration_osc633, span),
-                                    "reset_application_mode" => Value::bool(config.shell_integration_reset_application_mode, span),
-                                },
-                                span,
-                            );
-                        }
+                    } else {
+                        report_invalid_value("boolean value is deprecated, should be a record. see `config nu --default`.", span, &mut errors);
```

Comment:
With the new `ShellIntegationConfig` it should actually be quite tractable to again allow a bool here and simply set to `Default` on `false` and a sensible active config on `true`.